common/djangoapps/util/course.py

```diff
@@ -2,27 +2,49 @@
 Utility methods related to course
 """
 import logging
-from django.conf import settings
 
+from django.conf import settings
 from opaque_keys.edx.keys import CourseKey
 
+from openedx.core.djangoapps.catalog.utils import get_run_marketing_url
+
 log = logging.getLogger(__name__)
 
 
-def get_lms_link_for_about_page(course_key):
+def get_link_for_about_page(course_key, user):
     """
     Returns the url to the course about page.
     """
     assert isinstance(course_key, CourseKey)
 
     if settings.FEATURES.get('ENABLE_MKTG_SITE'):
-        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
-        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
-        about_base = settings.MKTG_URLS['ROOT']
-    else:
-        about_base = settings.LMS_ROOT_URL
+        marketing_url = get_run_marketing_url(course_key, user)
+        if marketing_url:
+            return marketing_url
+
+    return get_lms_course_about_page_url(course_key)
+
+
+def get_link_for_about_page_from_cache(course_key, catalog_course_run=None):
```

Comment:
@MatthewPiatetsky I believe we can remove assertion of CourseKey from `get_link_for_about_page_from_cache` as it only uses unicode value of `course_key`. I have updated the code in latest commit.

The only repeating code is the last 3 lines:
`if marketing_url:
return marketing_url`
and 
`return get_lms_course_about_page_url(course_key)`

I believe its worth keeping it as is instead of making the code more DRY by using any of the following method:
1. introducing more checks that would check if the call is intended for cache value or not (that would be to check values for `catalog_course_run` and passing `None` values for either `catalog_course_run` or `user` to handle both cases.)
or
2. creating another common function something like:
`def course_about_url(marketing_url):
return marketing_url if marketing_url else  get_lms_course_about_page_url(course_key)`
Which logically is not ideal as we would then return `marketing_url ` without any further operation on it if its not null.